src/config.c

```diff
@@ -5,13 +5,145 @@
 #include <stdlib.h>
 #include <ini.h>
 #include <error.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
 
 #include "log.h"
 #include "util.h"
 
+config_t *ini_config;
+char *log_filepath;
+extern FILE *log_file;
+extern char log_filename[256];
+
 #define CHECK_SECTION(s) strcmp(section, s) == 0
 #define MATCH(s, n) strcmp(section, s) == 0 && strcmp(name, n) == 0
 
+/*
+ *  construct_parent_directories()
+ *
+ *  Returns:
+ *  0 Success
+ * -1 Failure
+ */
+int create_parent_directories(char *path)
+{
+	/*
+	 * We use the `p` pointer to find each '/', then we temporary replace the '/'
+	 * character with '\0', so we can actually create the directory.
+	 * After that the '\0' gets replaced with '/' so we can continue iterating
+	 * through the path. The final directory won't be created this way, therefore
+	 * we have another call to `mkdir` after our `for` loop to make sure all the
+	 * directories are created.
+	 */
+	char *p;
+	for (p = path + 1; *p; p++) {
+		if (*p == '/') {
+			*p = '\0';
+			if (mkdir(path, 0777) != 0) {
+				if (errno != EEXIST) {
+					if (errno == EACCES) {
+						pr_error(
+							"Error creating directory %s, permission denied\n",
+							path);
+						return -1;
+					}
+				}
+			}
+			*p = '/';
+		}
+	}
+
+	if (mkdir(path, 0777) != 0) {
+		if (errno != EEXIST) {
+			if (errno == EACCES) {
+				pr_error(
+					"Error creating directory %s, permission denied\n",
+					path);
+				return -1;
+			}
+		}
+	}
+
+	// iterate to the end of the path so we can make sure to append '/' if needed
+	while (*p) {
+	}
+	if (*(p - 1) != '/') {
+		strncat(path, "/", strlen(path) + 1);
+	}
```

Comment:
This is for the case that the user put a path and forgot to append `/` at the end. For example:
`/path/to/log` would cause an error therefore we check and append to make it `/path/to/log/`.

Reply:
Move to log.

Reply:
You mean create a new function for that? Or move the whole `create_parent_directories` to log?

Reply:
Actually, all non-strictly-relevant functions to something need to be in utils. So `create_parent_directories` and friends need to be in util.c